src/panels/lovelace/special-rows/hui-call-service-row.ts

```diff
@@ -1,80 +1,98 @@
-import { html } from "@polymer/polymer/lib/utils/html-tag.js";
-import { PolymerElement } from "@polymer/polymer/polymer-element.js";
-import "@polymer/paper-button/paper-button.js";
-
-import "../../../components/ha-icon.js";
-import callService from "../common/call-service.js";
-
-class HuiCallServiceRow extends PolymerElement {
-  static get template() {
-    return html`
-      ${this.styleTemplate}
-      <ha-icon icon="[[_config.icon]]"></ha-icon>
-      <div class="flex">
-        <div>
-          [[_config.name]]
-        </div>
-        <paper-button on-click="_callService">[[_config.action_name]]</paper-button>
-      </div>
-    `;
-  }
-
-  static get styleTemplate() {
-    return html`
-      <style>
-        :host {
-          display: flex;
-          align-items: center;
-        }
-        ha-icon {
-          padding: 8px;
-          color: var(--paper-item-icon-color);
-        }
-        .flex {
-          flex: 1;
-          overflow: hidden;
-          margin-left: 16px;
-          display: flex;
-          justify-content: space-between;
-          align-items: center;
-        }
-        .flex div {
-          white-space: nowrap;
-          overflow: hidden;
-          text-overflow: ellipsis;
-        }
-        paper-button {
-          color: var(--primary-color);
-          font-weight: 500;
-          margin-right: -.57em;
-        }
-      </style>
-    `;
-  }
-
-  static get properties() {
-    return {
-      hass: Object,
-      _config: Object,
-    };
-  }
-
-  setConfig(config) {
-    if (
-      !config ||
-      !config.icon ||
-      !config.name ||
-      !config.action_name ||
-      !config.service ||
-      !config.service_data
-    ) {
-      throw new Error("Error in card configuration.");
-    }
-    this._config = config;
-  }
-
-  _callService() {
-    callService(this._config, this.hass);
-  }
-}
-customElements.define("hui-call-service-row", HuiCallServiceRow);
+import { html, LitElement } from "@polymer/lit-element";
+import "@polymer/paper-button/paper-button.js";
+
+import "../../../components/ha-icon.js";
+
+import callService from "../common/call-service.js";
+import { EntityRow, CallServiceConfig } from "../entity-rows/types.js";
+import { HomeAssistant } from "../../../types.js";
+import { TemplateResult } from "lit-html";
+
+class HuiCallServiceRow extends LitElement implements EntityRow {
+  public hass?: HomeAssistant;
+  private _config?: CallServiceConfig;
+
+  static get properties() {
+    return {
+      hass: {},
+      _config: {},
+    };
+  }
+
+  public setConfig(config: CallServiceConfig): void {
+    if (
+      !config ||
+      !config.icon ||

```

Comment:
We should do a default icon.

Reply:
Do you have a suggestion? I was thinking play-circle-outline

![play-circle-outline](https://user-images.githubusercontent.com/1287159/47666140-d15aa980-db70-11e8-839b-47c9fbaa497a.png)

Reply:
what icon do we use for services in the dev tools?